02-oop-inheritance/observerPattern.js

```diff
@@ -0,0 +1,140 @@
+//Inherit function
+var inherit = function inherit(child, parent){
+  var copyOfParent = Object.create(parent.prototype);
+  copyOfParent.constructor = child;
+ 	child.prototype = copyOfParent;	
+};
+
+//Augment function
+function augment(receivingClass, givingClass, methods) {
+  if (arguments[2]) {
+    for(var i = 0, length = methods.length; i < length; i++ ) {
+      receivingClass.prototype[methods[i]] = givingClass.prototype[methods[i]];
+    }
+  } else {
+    for (var methodName in givingClass.prototype ) {
+      if (!Object.hasOwnProperty.call(receivingClass.prototype, methodName)) {
+        receivingClass.prototype[methodName] = givingClass.prototype[methodName];
+      }
+    }
+  }
+};
+
+//Observer class
+var Observer = function Observer(events){
+	this.events = events || [];
+};
+
+//Observable class
+var Observable = function Observable(){
+	this.observersList = [];
+};
+
+Observable.prototype = {
```

Comment:
If you do this you should define the `constructor` property inside of the prototype definition